indexing-service/src/main/java/io/druid/indexing/common/task/CompactionTaskUtils.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.common.task;
+
+import com.google.common.base.Preconditions;
+import io.druid.java.util.common.guava.Comparators;
+import io.druid.segment.IndexIO;
+import io.druid.segment.QueryableIndex;
+import io.druid.segment.loading.SegmentLoadingException;
+import io.druid.timeline.DataSegment;
+import io.druid.timeline.TimelineObjectHolder;
+import io.druid.timeline.VersionedIntervalTimeline;
+import io.druid.timeline.partition.PartitionChunk;
+import io.druid.timeline.partition.PartitionHolder;
+import org.joda.time.Interval;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CompactionTaskUtils
+{
+  public static List<TimelineObjectHolder<String, DataSegment>> toTimelineSegments(
+      List<DataSegment> usedSegments,
+      Interval interval
+  ) throws IOException, SegmentLoadingException
+  {
+    final VersionedIntervalTimeline<String, DataSegment> timeline = new VersionedIntervalTimeline<>(
+        Comparators.naturalNullsFirst()
+    );
+
+    for (DataSegment segment : usedSegments) {
+      timeline.add(segment.getInterval(), segment.getVersion(), segment.getShardSpec().createChunk(segment));
+    }
+    return timeline.lookup(interval);
+  }
+
+  public static List<QueryableIndex> loadSegments(
```

Comment:
It looks like this helper method is only called in one spot. Does it really need to be in its own helper class?

Reply:
I tried to use this method in IngestSegmentFirehoseFactory as well, but it was not easy to do refactoring. Moved to CompactionTask now.